python/pyspark/pandas/series.py

```diff
@@ -4400,6 +4400,8 @@ def mode(self, dropna: bool = True) -> "Series":
         Series
             Modes of the Series.
 
+        .. note:: Since Spark 3.4.0, series name is preserved to follow pandas 1.4+ behavior.
```

Comment:
You can use `versionchanged` directive but I don't mind as is since it's already inconsistent.

Reply:
Sure, should we also mention these behavior changes in `docs/source/migration_guide/pyspark_3.3_to_3.4.rst`?

Reply:
Let's don't do it for now ..